server/api.js

```diff
@@ -1,15 +1,41 @@
-import express from "express";
+import express from "express"
+import * as db from "./data/models"
 
-const routerApi = express.Router();
+const routerApi = express.Router()
 
-//mongodb models
-import * as db from "./data/models";
-
-//routes
+// DEALS - START
 routerApi.get("/deals", async (req, res) => {
-    db.dealsCollection.find().exec(function(err, result) {
-        res.send(result);
-    });
-});
+  let { title } = req.query
+
+  if (title) {
+    await db.dealsCollection.findOne(req.query).exec((err, result) => res.send(result))
+  } else {
+    await db.dealsCollection.find().exec((err, result) => res.send(result))
+  }
```

Comment:
If we change these 5 lines to:
`await db.dealsCollection.find(req.query).exec((err, result) => res.send(result))`
Will this change produce the same result?